File: ecommerce_api/src/main/java/personal/rajit/controller/CustomerController.java
package personal.rajit.controller;

import lombok.AllArgsConstructor;
import org.springframework.data.domain.Page;
import org.springframework.http.HttpStatusCode;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;
import personal.rajit.controller.request_dto.CustomerSearchDto;
import personal.rajit.entity.Customer;
import personal.rajit.service.CustomerService;

@RestController
@RequestMapping("/customer")
@AllArgsConstructor
public class CustomerController {

    private final CustomerService customerService;

    @PostMapping("/save")
    public ResponseEntity<Customer> save(@RequestBody Customer customer) {
        System.out.println(customer.getId());
        System.out.println(customer.getName());
        return new ResponseEntity(customerService.saveCustomer(customer), HttpStatusCode.valueOf(200));
    }

    @PostMapping("/update")
    public ResponseEntity<Customer> update(@RequestBody Customer customer) {
        return new ResponseEntity(customerService.updateCustomer(customer), HttpStatusCode.valueOf(200));
    }

    @PostMapping("/delete")
    public ResponseEntity<String> deleteCustomer(@RequestBody Customer customer) {
        return new ResponseEntity(customerService.deleteCustomer(customer), HttpStatusCode.valueOf(200));
    }

    @PostMapping("/search")
    public ResponseEntity<Page<Customer>> search(@RequestBody CustomerSearchDto customerSearchDto) {
        return new ResponseEntity(customerService.searchCustomer(customerSearchDto), HttpStatusCode.valueOf(200));
    }

    @PostMapping("/make-active")
    public ResponseEntity<Page<Customer>> makeActive(@RequestBody Customer customer) {
        return new ResponseEntity(customerService.makeActiveCustomer(customer), HttpStatusCode.valueOf(200));
    }
}

File: frontend/src/app/controller/CustomerController.ts
import { HttpClient } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { Observable } from "rxjs";
import { Customer } from "../entity/Customer";
import { CustomerSearchDto } from "./search_dto/CustomerSearchDto";
import { Page } from "../dto/Page";

@Injectable({ providedIn: 'root' })
export class CustomerController {

  constructor(private httpClient: HttpClient) { }

  save(customer: Customer): Observable<Customer> {
    return this.httpClient.post<Customer>('http://localhost:8080/customer/save', 
    customer);
  }

  update(customer: Customer): Observable<Customer> {
    return this.httpClient.post<Customer>('http://localhost:8080/customer/update', 
    customer);
  }

  delete(customer: Customer): Observable<boolean> {
    return this.httpClient.post<boolean>('http://localhost:8080/customer/delete', 
    customer)
  }

  search(customerSearchDto: CustomerSearchDto): Observable<Page<Customer>> {
    return this.httpClient.post<Page<Customer>>('http://localhost:8080/customer/search', 
    customerSearchDto);
  }

  makeActive(customer: Customer) {
    return this.httpClient.post<boolean>('http://localhost:8080/customer/make-active', 
    customer)
  }

}
